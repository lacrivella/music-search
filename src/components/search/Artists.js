import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

function Artists({ artists }) {
  const artistList = artists.map(artist => {
    return (
      <li key={artist.id}>
        <Artist artist={artist} />
      </li>
    );
  });
  return (
    <ul>
      {artistList}
    </ul>
  );
}

Artists.propTypes = {
  artists: PropTypes.array.isRequired
};

export default Artists;
