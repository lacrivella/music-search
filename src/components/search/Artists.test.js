import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists';

describe('Artists component', () => {
  it('renders Artists', () => {
    const wrapper = shallow(<Artists artists={[{ name: 'WASP', disambiguation: 'Metal Band', id: 'whatev' }, {name: 'Poison', disambiguation: 'Hair Metal Band', id: 'meep' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});