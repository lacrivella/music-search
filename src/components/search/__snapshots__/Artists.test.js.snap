// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists component renders Artists 1`] = `
<ul>
  <li
    key="whatev"
  >
    <Artist
      artist={
        Object {
          "disambiguation": "Metal Band",
          "id": "whatev",
          "name": "WASP",
        }
      }
    />
  </li>
  <li
    key="meep"
  >
    <Artist
      artist={
        Object {
          "disambiguation": "Hair Metal Band",
          "id": "meep",
          "name": "Poison",
        }
      }
    />
  </li>
</ul>
`;
